hydra-node/test/Hydra/Chain/Direct/StateSpec.hs

```diff
@@ -404,6 +415,29 @@ forAllNonEmptyByronCommit action = do
           Right{} -> property False
           Left e -> action e
 
+forAllCommitWithMoreThan100ADA ::
+  (PostTxError Tx -> Property) ->
+  Property
+forAllCommitWithMoreThan100ADA action = do
+  forAll (genHydraContext maximumNumberOfParties) $ \hctx ->
+    forAll (genStInitial hctx) $ \(ctx, stInitial) ->
+      forAll genAdaOnlyUTxOOnMainnetWithAmountBiggerThan100k $ \utxo ->
+        case commit (alterChainContextNetworkToMainnet ctx) stInitial utxo of
+          Right{} -> property False
+          Left e -> action e
+ where
+  genAdaOnlyUTxOOnMainnetWithAmountBiggerThan100k :: Gen UTxO
+  genAdaOnlyUTxOOnMainnetWithAmountBiggerThan100k = do
+    vk <- arbitrary
+    adaAmount <- arbitrary `suchThat` (> 0)
+    let value = lovelaceToValue $ Lovelace (adaAmount + 100_000_000)
+    txIn <- arbitrary
+    pure . UTxO.UTxO $
+      Map.singleton txIn (TxOut (mkVkAddress Mainnet vk) value TxOutDatumNone ReferenceScriptNone)
```

Comment:
question: can we have more output entries as part of the UTxO, which value sum > limit?

here, we are building a map with a single entry using an over-the-limit value.
is it possible to have a UTx0 with multiple outputs of small values whose sum is greater than the limit?

Reply:
Yes, but this will trigger a `MoreThanOneUTxOCommitted` IIRC. So having a single entry generator is fine here.

I would suggest to use `genUTxO1` which I created here: https://github.com/input-output-hk/hydra/pull/766/files#diff-a888f29e34de2b98f0f64ed4ec9d6e396ef5a5ae992398fea56bac95eb7d58aeR274